test/integration/challenges.test.js

```diff
@@ -0,0 +1,111 @@
+const chai = require('chai')
+const { expect } = chai
+const chaiHttp = require('chai-http')
+
+const app = require('../../server')
+const authService = require('../../services/authService')
+const addUser = require('../utils/addUser')
+const cleanDb = require('../utils/cleanDb')
+
+const config = require('config')
+const cookieName = config.get('userToken.cookieName')
+
+chai.use(chaiHttp)
+
+describe('Challenges', function () {
+  let jwt
+
+  beforeEach(async function () {
+    const userId = await addUser()
+    jwt = authService.generateAuthToken({ userId })
+  })
+
+  afterEach(async function () {
+    await cleanDb()
+  })
+
+  describe('GET /challenges', function () {
+    it('Should return the available challenges', function (done) {
+      chai
+        .request(app)
+        .get('/challenges')
+        .end((err, res) => {
+          if (err) {
+            return done(err)
+          }
+          expect(res).to.have.status(200)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Challenges returned successfully!')
+          expect(res.body.challenges).to.be.a('array')
+
+          return done()
+        })
+    })
+    it('Should return a not found message if the challenges is not found', function (done) {
```

Comment:
Sure about this? Is the code written this way? 

If yes, it shouldn't be. 4xx errors are for client's mistake. If there's no challenge in DB, that isn't user's fault.

Reply:
Yes, I've checked routes for challenges and it had 404 only.

Reply:
@Ishikabaid we don't return 404 for 'No challenges'. You can see [here](https://github.com/Real-Dev-Squad/website-backend/blob/5ec05e86695a2e717030156e91f779f45c611adf/controllers/challenge.js?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L17). We check whether `challenges` variable is null or not, and according to that we return the msg and challenges.